kernel/src/process/process_vm/init_stack/mod.rs

```diff
@@ -446,7 +457,14 @@ impl InitStackReader<'_> {
                     break;
                 }
 
-                let mut envp_reader = frame.reader().skip(envp_ptr - page_base_addr).to_fallible();
+                let mut envp_reader =
+                    frame
+                        .reader()
+                        .skip(envp_ptr.checked_sub(page_base_addr).ok_or_else(|| {
+                            Error::with_message(Errno::EINVAL, "envp is corrupted")
+                        })?)
```

Comment:
Define a temporary variable with a self-describing name to make the code more readable.

```rust
let envp_offset = envp_ptr.checked_sub(page_base_addr)
    .ok_or_else(|| {
        Error::with_message(Errno::EINVAL, "envp is corrupted")
    })?;
let mut envp_reader = frame.reader().skip(envp_offset).to_fallible();
```

The changes to `arg_reader` can be improved similarly.